### PR TITLE
Add loadout handling + some UI for SRO

### LINKED ISSUE #2190: Add loadout context + handling

### PR DESCRIPTION
**Pull request by nguyentvan7**
## Describe your changes

* Remove usage of LocationCharacterKey, since each trailblazer path is a distinct character. Hurray for huge system simplification
* Refactor CalcProvider to be a TeamCalcProvider, accepting a teamId
* Delete all content from home page (now in the team page)
* Rename TeamChar -> LoadoutData
* Rename LoadoutDatum -> LoadoutMetadatum
* Add TeamCalcProvider to team page, with support for character swapping/teammate handling, etc
* Add CharacterSelector to TeamSettings to allow creation of teams

## Issue or discord link

- Resolve #2190

## Testing/validation

chrome_t1Ny9tpmlj

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: libs/sr/db/src/Database/DataManagers/BuildDataManager.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export interface Build {
   description: string
 
   lightConeId?: string
-  relicIds: Record<RelicSlotKey, string | undefined>
```
Comment (frzyc) [inline libs/sr/db/src/Database/DataManagers/BuildDataManager.ts:13]: would prefer if we go by `string | null` for Ids for the future to better adhere to a nullable field in a relational database.

File: libs/sr/ui/src/Character/LocationAutocomplete.tsx [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
+import { allCharacterKeys } from '@genshin-optimizer/sr/consts'
 import { Skeleton } from '@mui/material'
 import { Suspense, useCallback, useMemo } from 'react'
 import { useTranslation } from 'react-i18next'
 import { useDatabaseContext } from '../Context'
 
 type LocationAutocompleteProps = {
-  locKey: CharacterLocationKey | ''
-  setLocKey: (v: CharacterLocationKey | '') => void
+  locKey: CharacterKey | ''
```
Comment (frzyc) [inline libs/sr/ui/src/Character/LocationAutocomplete.tsx:11]: does not have to be this PR, but I would prefer `CharacterKey | null` here as well.

**Review by frzyc — APPROVED** [review]
minor nits, mostly about `'' -> null` stuff, I am fine with merging this in for now and address that later.

File: libs/sr/db/src/Database/DataManagers/BuildDataManager.ts (hunk 1 above)
Comment (nguyentvan7) [reply libs/sr/db/src/Database/DataManagers/BuildDataManager.ts:13]: https://github.com/users/frzyc/projects/2?pane=issue&itemId=76062503

File: libs/sr/ui/src/Character/LocationAutocomplete.tsx (hunk 2 above)
Comment (nguyentvan7) [reply libs/sr/ui/src/Character/LocationAutocomplete.tsx:11]: https://github.com/users/frzyc/projects/2?pane=issue&itemId=76062503

**Pull request merged**